include/swift/SIL/SILInstruction.h

```diff
@@ -2133,6 +2138,32 @@ class ApplyInstBase<Impl, Base, true>
   bool hasSemantics(StringRef semanticsString) const {
     return doesApplyCalleeHaveSemantics(getCallee(), semanticsString);
   }
+
+// SWIFT_ENABLE_TENSORFLOW
+private:
+  /// Predicate used to filter InoutArgumentRange.
+  struct OperandToInoutArgument {
+    const Impl &inst;
+    OperandToInoutArgument(const Impl &inst) : inst(inst) {}
+    Optional<SILValue> operator()(unsigned long i) const {
+      auto paramInfo = inst.getSubstCalleeConv().getParameters()[i];
```

Comment:
Maybe store these two ranges as properties in the struct? `inst.getSubstCalleeConv().getParameters()` and `inst.getArgumentsWithoutIndirectResults()`.

Reply:
Done in https://github.com/apple/swift/pull/28352/commits/d43ec714c6f0bb34b00801b14ba29d69ef63fc43.